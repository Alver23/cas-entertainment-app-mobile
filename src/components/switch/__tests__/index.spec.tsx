// Dependencies
import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import SwitchSelector from 'react-native-switch-selector';

import Switch from '../index';

describe('<Switch />', () => {
  let component: ShallowWrapper;

  const props = {
    onPress: jest.fn(),
    options: [],
    initialOption: 0,
  };

  beforeEach(() => {
    component = shallow(<Switch {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the onPress void', () => {
    const onPress: any = component.find(SwitchSelector).prop('onPress');
    onPress();
    expect(props.onPress).toHaveBeenCalledTimes(1);
  });
});
