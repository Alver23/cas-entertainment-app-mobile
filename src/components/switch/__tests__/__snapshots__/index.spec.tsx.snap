// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render correctly 1`] = `
<SwitchSelector
  animationDuration={350}
  backgroundColor="#FFFFFF"
  bold={false}
  borderColor="#C9C9C9"
  borderRadius={8}
  borderWidth={1}
  buttonColor="#2089dc"
  buttonMargin={0}
  disableValueChangeOnPress={false}
  disabled={false}
  fontSize={14}
  hasPadding={false}
  height={28}
  imageStyle={Object {}}
  initial={0}
  onPress={[MockFunction]}
  options={Array []}
  returnObject={false}
  selectedColor="#FFFFFF"
  selectedTextContainerStyle={Object {}}
  selectedTextStyle={Object {}}
  style={
    Object {
      "borderColor": "#2089dc",
      "borderRadius": 8,
      "borderWidth": 1,
      "width": 360,
    }
  }
  textColor="#000000"
  textContainerStyle={Object {}}
  textStyle={Object {}}
  valuePadding={1}
/>
`;
